src/Service/Resolvers/MsSqlQueryExecutor.cs

```diff
@@ -186,13 +187,22 @@ public override string GetSessionParamsQuery(HttpContext? httpContext, IDictiona
             foreach ((string claimType, Claim claim) in sessionParams)
             {
                 string paramName = $"{SESSION_PARAM_NAME}{counter.Next()}";
-                parameters.Add(paramName, claim.Value);
+                parameters.Add(paramName, new(claim.Value));
                 // Append statement to set read only param value - can be set only once for a connection.
                 string statementToSetReadOnlyParam = "EXEC sp_set_session_context " + $"'{claimType}', " + paramName + ", @read_only = 1;";
                 sessionMapQuery = sessionMapQuery.Append(statementToSetReadOnlyParam);
             }
 
             return sessionMapQuery.ToString();
         }
+
+        /// <inheritdoc/>
+        public override void PopulateDbTypeForParameter(KeyValuePair<string, DbConnectionParam> parameterEntry, DbParameter parameter)
+        {
+            if (parameterEntry.Value.DbType is not null)
```

Comment:
are we certain parameterEntry.Value is not null? shouldnt we check for it first?

Reply:
Yeah, actually we are sure that the parameter.Value is not going to be null if you refer to the only usage of this function here:
https://github.com/Azure/data-api-builder/blob/4c3b07324f9e24dae821a2c5d7853ed73835b7b2/src/Service/Resolvers/QueryExecutor.cs#L166
which is inside a traversal performed on all the parameters. We can still however use another null check since this method is public.